crates/core/component/dex/src/component/router/route_and_fill.rs

```diff
@@ -79,34 +96,25 @@ pub trait HandleBatchSwaps: StateWrite + Sized {
             trading_pair,
             delta_1,
             delta_2,
-            lambda_1,
-            lambda_2,
-            unfilled_1,
-            unfilled_2,
+            lambda_1: swap_execution_2_for_1.output.amount,
+            lambda_2: swap_execution_1_for_2.output.amount,
+            unfilled_1: delta_1 - swap_execution_1_for_2.input.amount,
+            unfilled_2: delta_2 - swap_execution_2_for_1.input.amount,
         };
 
         // TODO: how does this work when there are trades in both directions?
         // Won't that mix up the traces? Should the SwapExecution be indexed by
         // the _DirectedTradingPair_?
 
         // Fetch the swap execution object that should have been modified during the routing and filling.
-        let trade_traces: im::Vector<Vec<Value>> = self
-            .object_get("trade_traces")
-            .ok_or_else(|| anyhow::anyhow!("missing swap execution in object store2"))?;
-        tracing::debug!(?output_data, ?trade_traces);
+        tracing::debug!(
+            ?output_data,
+            ?swap_execution_1_for_2,
+            ?swap_execution_2_for_1
+        );
         Arc::get_mut(self)
             .expect("expected state to have no other refs")
-            .set_output_data(
-                output_data,
-                SwapExecution {
-                    traces: trade_traces.into_iter().collect(),
-                },
-            );
-
-        // Clean up the swap execution object store now that it's been persisted.
-        Arc::get_mut(self)
-            .expect("expected state to have no other refs")
-            .object_delete("trade_traces");
+            .set_output_data(output_data, swap_execution_1_for_2, swap_execution_2_for_1);
```

Comment:
Probably we want to decouple setting the BSOD from the `SwapExecution`s, so that we can write the `SwapExecution` only in the branch where we actually execute it.